### DISCUSSION TITLE
Passing cli arguments and normal arguments in same function

### DISCUSSION THREAD

**Discussion by therc01** [fastapi/typer#1320] [Questions]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Currently this feature is not present in typer.
```

### Description

I want to pass cli arguments and normal arguments to a function using typer. 

### Wanted Solution

A way to pass arguments from command lines and normal arguments different than cli arguments to same function. 

### Wanted Code

```python
example code:

import typer

def simple( argument1, argument2: str):
    print(argument1, argument2)

if __name__ == '__main__':
    with open(file, 'r') as f:
        argument1 = json.load(f)
    typer.run(work(argument1))
```

### Operating System

Linux

### Typer Version

0.4.0

### Python Version

Python 3.8.10

**Comment by YuriiMotov** [comment]
[Having the support of `partial`](https://github.com/fastapi/typer/discussions/699) would solve this:
```py
    typer.run(
        functools.partial(simple, argument1)
    )
```

But it's not supported..

You can try the workaround with `makefun`: create a tool that will work like `functool.partial`, but the result will be a real function with updated signature.

<details><summary>Look in the details</summary>

```py
import functools
import inspect
import json
from typing import Any

import typer
from makefun import create_function  # pip install makefun

def my_partial(func, *pargs, **pkwargs):
    """
    Like functools.partial, but returns a real function with
    substituted parameters removed from the signature.
    """
    p = functools.partial(func, *pargs, **pkwargs)
    sig = inspect.signature(func)
    bound = sig.bind_partial(*pargs, **pkwargs)

    # Keep only the parameters that were not fixed
    remaining_params = [
        param for name, param in sig.parameters.items()
        if name not in bound.arguments
    ]
    new_sig = "(" + ", ".join(str(param) for param in remaining_params) + ")"

    def wrapper(*args, **kwargs):
        return p(*args, **kwargs)

    return create_function(
        new_sig,
        wrapper,
        func_name=func.__name__,
        doc=func.__doc__,
        annotations=func.__annotations__,
    )

def simple(
    argument1: Any,
    argument2: str,
):
    print(argument1, argument2)

if __name__ == "__main__":
    with open("file.json") as f:
        argument1 = json.load(f)

    typer.run(my_partial(simple, argument1))

```

</details>
